packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -179,6 +179,7 @@ class Draggable<T extends Object> extends StatefulWidget {
     this.ignoringFeedbackPointer = true,
     this.rootOverlay = false,
     this.hitTestBehavior = HitTestBehavior.deferToChild,
+    this.allowListPointer = const <int>[kPrimaryButton]
```

Comment:
Design: If `this.allowListPointer` now has a default value, then all clients of this API will be opted-in. I think that we eventually want to do this, but not yet. 

[Flutter's compatibility policy](https://docs.flutter.dev/resources/compatibility) does not qualify this as a breaking change because it does not break existing tests. However, I think that we should treat it as such since the behavior of apps will change.

For now, I suggest that we allow this value to be null and make a new issue to track widgets where we make similar changes. How does that sound?

- [ ] Action Item: Make the following change.
```dart
this.allowListPointer,
```

Reply:
Another idea here is that the framework should allow the user to determine the predicate (what is allowed) for the `PointerDownEvent`. 

```dart
// default the value in the parameter list
// we default it here so that our changes are backward compatible
this.isPointerAllowed = (PointerDownEvent event) => true,

// where it is defined as a property in the class
final bool Function(PointerDownEvent)? isPointerAllowed;
```

Right now the way that this change checks whether we allow this PointerDownEvent is 
```dart
if (widget.allowListPointer == null || widget.allowListPointer!.contains(event.buttons))
```
but what if the `PointerDownEvent` contains both `kPrimaryButton & kSecondaryButton`? 

The above code suggests that this gesture is allowed, which is heavy-handed considering that users might want to ignore that. 

This example should illustrate how using a predicate might be more advantageous rather than an allow list.

Reply:
> For now, I suggest that we allow this value to be null and make a new issue to track widgets where we make similar changes. How does that sound?

Yes, I can make this. But... I don't think any other widget is affected. Sure, it might be, I agree we can study about this, but from my head, most of them use GestureDetector or something else. This one is using bare-bones Listener, so it behaves as an exception.

Reply:
> what if the PointerDownEvent contains both kPrimaryButton & kSecondaryButton?

There is a test for this. The allowList allows anything inside. If you have kPrimary and kSecondary, it would work for any of them. But would fail for Scroll-click. If you had [kPrimary, kSecondary, kScrollClick(?)], then it would behave like right now.

Reply:
This is more radical, in Listener:
![image](https://user-images.githubusercontent.com/351125/191351885-ea50b906-5a57-44d4-8d03-3541fa44510e.png)

Reply:
These are all the Listener usages inside /widgets. Most are related to scrolling.

<img width="618" alt="image" src="https://user-images.githubusercontent.com/351125/191354020-b4c6b227-5db9-4bee-953f-222d19c0460e.png">
<img width="694" alt="image" src="https://user-images.githubusercontent.com/351125/191354097-835acd57-7033-4a09-96e5-ffd466c2179d.png">

<img width="484" alt="image" src="https://user-images.githubusercontent.com/351125/191354558-71a469d3-e987-4292-8315-77a74638b387.png">

Reply:
> Another question I have is: what if we "fixed" Listener to only allow specific buttons?

That's a more impactful solution that we should only take if it's absolutely necessary. If you can justify a reason why it should be addressed there, we can consider it. Otherwise, we should narrow the scope of handling `PointerDownEvents` with the "right-click only" behavior to `Draggable.`

Reply:
> Yeah, I'm not sure I like the predicate thing, probably the allow list concept is easier to understand.

I'll give more context and an elaborate example. Let me know if this helps.

## From [`PointerDownEvent.buttons`](https://api.flutter.dev/flutter/gestures/PointerEvent/buttons.html)  
> Bit field using the *Button constants such as [kPrimaryMouseButton](https://api.flutter.dev/flutter/gestures/kPrimaryMouseButton-constant.html), [kSecondaryStylusButton](https://api.flutter.dev/flutter/gestures/kSecondaryStylusButton-constant.html), etc.
> 
> For example, if this has the value 6 and the [kind](https://api.flutter.dev/flutter/gestures/PointerEvent/kind.html) is [PointerDeviceKind.invertedStylus](https://api.flutter.dev/flutter/dart-ui/PointerDeviceKind.html), then this indicates an upside-down stylus with both its primary and secondary buttons pressed.

`PointerDownEvent.buttons` can be composed of many buttons. 

Now, suppose that we just have two buttons. For example, `kPrimaryButton = (0x01)/(0001)` and `kSecondaryButton = (0x02)/(0010)`. If the `PointerDownEvent` registered a click from `kPrimaryButton` and `kSecondaryButton` its value will be `kPrimaryButton + kSecondaryButton = 4`. If our allowlist values contains `[kPrimaryButton, kSecondaryButton]`

This condition 
```dart
widget.allowListPointer!.contains(event.buttons))
```
is false since `[1, 2]` does not contain `3`. But, I think you mean for it to be true.

Reply:
Oh, crap. I hate these. I thought they were enum (maybe we can make them enhanced enum now? 👀). You are right, the allowList wouldn't work for this.
But `kPrimary in [kPrimary, kSecondary]` would work, just not `kPrimary` in `kPrimary + kSecondary`.

Reply:
So, the allowList could be `int? allowButtons` How do you feel? 0 = none. Null = all.

Reply:
> So, the allowList could be int? allowButtons How do you feel? 0 = none. Null = all.

That's definitely another approach that we could take here, but that would defer the calculation of `allowButtons` via bitwise operations to the client of this API. The goal here should be to make the API as simple as possible to use, and I think we have an opportunity here to hide some of the messy details of `PointerDownEvent.buttons` and programmability via something like `allowList` accessible to beginners.

Reply:
Thanks for all your helpful comments. I just pushed some great changes. Now I'll see how far I can get with the bitwise you shared with me 😀

Reply:
Your algorithm works!!!! That's fantastic. lol.

Reply:
I need to work now. I'll check what's wrong with the tests tomorrow. Feel free to check if you like the current state :)

See you later! Thanks for all the help!